tests/unit/amo/components/TestRatingManager.js

```diff
@@ -84,471 +103,321 @@ describe(__filename, () => {
-    const { store } = dispatchSignInActions();
-    const dispatchSpy = sinon.spy(store, 'dispatch');
-
-    const errorHandler = createStubErrorHandler(new Error('unexpected error'));
-
-    const props = getRenderProps({
-      addon,
-      dispatch: store.dispatch,
-      errorHandler,
+    const addon = createInternalAddonWithLang(fakeAddon);
+    dispatchSignInActionsWithStore({ store, userId: defaultUserId });
+    createFailedErrorHandler({
+      id: errorHandlerId,
       store,
     });
-    const mapped = mapStateToProps(store.getState(), props);
 
-    // This uses shallow() because shallowUntilTarget() cannot handle
-    // the <div /> returned from withRenderedErrorHandler().
-    // It throws:
-    // ShallowWrapper::dive() can not be called on Host Components
-    shallow(<RatingManagerBase {...props} {...mapped} />);
+    const dispatch = jest.spyOn(store, 'dispatch');
+    render({ addon });
 
-    sinon.assert.notCalled(dispatchSpy);
+    expect(dispatch).not.toHaveBeenCalledWith(
+      expect.objectContaining({ type: FETCH_LATEST_USER_REVIEW }),
+    );
   });
 
   it('passes review=undefined before the saved review has loaded', () => {
-    const addon = createInternalAddonWithLang({ ...fakeAddon });
-    const { store } = dispatchSignInActions();
-
-    const root = render({ addon, store });
+    dispatchSignInActionsWithStore({ store, userId: defaultUserId });
+    render();
 
-    expect(root.find(UserRating)).toHaveProp('review', undefined);
+    expect(screen.getByRole('alert')).toBeInTheDocument();
+    expect(
+      screen.queryByTitle('There are no ratings yet'),
+    ).not.toBeInTheDocument();
   });
 
   it('passes review=null if no user is signed in', () => {
-    const { store } = dispatchClientMetadata();
-    const root = render({ store });
+    render();
 
+    expect(screen.getAllByTitle('There are no ratings yet')).toHaveLength(6);
     // This does not trigger a loading state.
-    expect(root.find(UserRating)).toHaveProp('review', null);
+    expect(screen.queryByRole('alert')).not.toBeInTheDocument();
   });
 
   it('passes the review once it has loaded', () => {
-    const externalReview = { ...fakeReview, id: 432 };
-    const { addon, store } = createStoreWithLatestReview({
-      review: externalReview,
-    });
-    const root = render({ addon, store });
+    renderWithReview({ review: { ...fakeReview, score: 1 } });
 
-    const rating = root.find(UserRating);
-    expect(rating).toHaveProp('review', createInternalReview(externalReview));
+    expect(screen.getByTitle('Rated 1 out of 5')).toBeInTheDocument();
   });
 
   it('passes review=null when no saved review exists', () => {
-    const { addon, store } = createStoreWithLatestReview({ review: null });
-    const root = render({ addon, store });
+    renderWithReview({ review: null });
 
     // This exits the loading state.
-    expect(root.find(UserRating)).toHaveProp('review', null);
-  });
-
-  it('configures a rating component', () => {
```

Comment:
Note that this test has been replaced with other tests that verify and interact with Ratings.